ibm_catalog.json

```diff
@@ -1,445 +1,801 @@
+      "product_kind": "solution",
+      "tags": [
+        "dev_ops",
+        "target_terraform",
+        "terraform",
+        "solution",
+        "storage",
+        "ibm_created"
+      ],
+      "keywords": [
+        "terraform",
+        "cos",
+        "bucket",
+        "storage",
+        "solution"
+      ],
+      "short_description": "Creates and configures an IBM Cloud Object Storage instance and bucket",
+      "long_description": "This architecture supports creating an IBM Cloud Object Storage instance, cross-region and single-region bucket.",
+      "offering_docs_url": "https://github.com/terraform-ibm-modules/terraform-ibm-cos/blob/main/README.md",
+      "offering_icon_url": "https://raw.githubusercontent.com/terraform-ibm-modules/terraform-ibm-cos/main/images/cos_icon.svg",
+      "provider_name": "IBM",
+      "features": [
+        {
+          "title": "Create an IBM Cloud Object Storage instance",
+          "description": "Create an IBM Cloud Object Storage instance."
+        },
+        {
+          "title": "Create an IBM Cloud Object Storage bucket",
+          "description": "Create either a cross-region or regional Object Storage bucket."
+        },
+        {
+          "title": "Attach access tags",
+          "description": "Attach access tags to the Object Storage instance."
+        }
+      ],
+      "flavors": [
+        {
+          "label": "Single instance",
+          "name": "instance",
+          "install_type": "fullstack",
+          "working_directory": "solutions/instance",
+          "compliance": {
+            "authority": "scc-v3",
+            "profiles": [
+              {
+                "profile_name": "IBM Cloud Framework for Financial Services",
+                "profile_version": "1.7.0"
+              }
+            ]
+          },
+          "architecture": {
             "features": [
+              {
+                "title": "Create an IBM Cloud Object Storage instance",
+                "description": "Create and configure an IBM Cloud Object Storage instance."
+              },
+              {
+                "title": "Create an IBM resource key",
+                "description": "Create a resource key for the Object Storage instance."
+              }
+            ],
+            "diagrams": [
+              {
+                "diagram": {
+                  "caption": "IBM Cloud Object Storage Solution",
+                  "url": "https://raw.githubusercontent.com/terraform-ibm-modules/terraform-ibm-cos/main/reference-architectures/instance.svg",
+                  "type": "image/svg+xml"
+                },
+                "description": "This architecture supports creating and configuring an IBM Cloud Object Storage instance."
+              }
+            ]
+          },
+          "configuration": [
+            {
+              "key": "ibmcloud_api_key",
+              "required": true,
+              "type": "password"
```

Comment:
No need to add a type here - it will default to a password type since the terraform variable is marked as secret